docs/blueprints/log-in-with-storj.md

```diff
@@ -0,0 +1,205 @@
+- `object:delete` - optional, allows deleting object data
+
+**What are they allowed to do it to?**
+
+These additional scopes are added by the authenticating user to restrict an applications access to a single project and
+bucket.
+
+- `project:<project>`  - required, scopes operations to a single project
+- `bucket:<name>`      - required, scopes operations to a selected bucket
+- `key:<algo>:<key>`   - required, communicates the encryption key to be used for communication
+
+**Access Tokens, Refresh Tokens, and API Keys**
+
+OAuth2 and OIDC do not require that access and refresh tokens be formatted a specific way (i.e. JWTs). In OAuth, access
+tokens are short-lived tokens that allow an application to act on behalf of an end user for a limited period of time
+(i.e. a bearer token). Refresh tokens are optionally issued, allowing applications to refresh the user identity for some
+longer period of time. The idea is to minimize the amount of time that a given token is valid for. Consider the
+following diagram that shows how a malicious attacker might obtain an access token from a third-party application or
+even from the user's browser cache.
+
+![attack](images/log-in-with-storj-attack.svg)
+
+Due to the lack of specificity in the format of access and refresh tokens, this allows us to conveniently plug in our
+macaroons (especially since there are many similarities between them and JWTs). One nice element that we get from
+macaroons is the ability to generate subsequent access tokens from the refresh token. Consider the following Golang
+snippet that demonstrates this:
+
+```go
+apiKey, _ := macaroon.NewAPIKey(projectSecret)
+apiKey, _ = apiKey.Restrict(grants)              // explicit grants from the consent page
+
+refreshToken, _ := apiKey.Restrict(refreshTokenTimeWindow) // include a nonce !important
+accessToken, _ := refreshToken.Restrict(accessTokenTimeWindow) // include a nonce !important
+```
+
+When a valid refresh token is presented for renewing, we can skip some expensive API calls and simply mint a new access
+token from the provided refresh token. The newly minted access token can then be used to communicate with the Storj API.
+
+**Encryption Key and Algorithm**
+
+One thing that our workflow needs to account for that OIDC and OAuth2 do not explicitly detail is the handoff of an
+encryption key without exposing the root a passphrase. One way that this can be done is when Alice is authorizing
+StorJIF to manage objects in Storj in step 4 above, we require that they enter the required passphrase before submitting
+the form. On submit, we can use that passphrase to generate an encryption key and pass that back to the server to ensure
+the passphrase is never exposed to the Storj ecosystem. The resulting key and algorithm can be cached along with the
+profile or "session" data so when StorJIF requests the user information in step 7, they can obtain the associated
+encryption key.
+
+<!--
```

Comment:
I think is critical from user perspective. While granting access we definitely need to be clear about when the external server has access to the unencrypted content.

Reply:
Do you think it's worth explicitly calling out to this on our consent screen?

Reply:
Yes. There's a reason we are doing the same in our satellite object browser.